presto-main/src/main/java/com/facebook/presto/operator/scalar/ApplyFunction.java

```diff
@@ -0,0 +1,94 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.metadata.BoundVariables;
+import com.facebook.presto.metadata.FunctionKind;
+import com.facebook.presto.metadata.FunctionRegistry;
+import com.facebook.presto.metadata.Signature;
+import com.facebook.presto.metadata.SqlScalarFunction;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeManager;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+
+import java.lang.invoke.MethodHandle;
+
+import static com.facebook.presto.metadata.Signature.typeVariable;
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static com.google.common.primitives.Primitives.wrap;
+
+public final class ApplyFunction
+        extends SqlScalarFunction
+{
+    public static final ApplyFunction APPLY_FUNCTION = new ApplyFunction();
+
+    private static final MethodHandle METHOD_HANDLE = methodHandle(ApplyFunction.class, "apply", MethodHandle.class, Object.class);
+
+    private ApplyFunction()
+    {
+        super(new Signature(
+                "apply",
+                FunctionKind.SCALAR,
+                ImmutableList.of(typeVariable("T"), typeVariable("U")),
+                ImmutableList.of(),
+                parseTypeSignature("U"),
+                ImmutableList.of(parseTypeSignature("function(T,U)"), parseTypeSignature("T")),
+                false));
+    }
+
+    @Override
+    public boolean isHidden()
+    {
+        return true;
+    }
+
+    @Override
+    public boolean isDeterministic()
+    {
+        return true;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "lambda apply function";
+    }
+
+    @Override
+    public ScalarFunctionImplementation specialize(BoundVariables boundVariables, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        Type argumentType = boundVariables.getTypeVariable("T");
+        Type returnType = boundVariables.getTypeVariable("U");
+        return new ScalarFunctionImplementation(
+                true,
+                ImmutableList.of(false, true),
+                METHOD_HANDLE.asType(
+                        METHOD_HANDLE.type()
+                                .changeReturnType(wrap(returnType.getJavaType()))
+                                .changeParameterType(1, wrap(argumentType.getJavaType()))),
+                isDeterministic());
+    }
+
+    public static Object apply(MethodHandle function, Object input)
+    {
+        try {
+            return function.invoke(input);
```

Comment:
It works. I added test to cover this case. Right now, lambdas are always generated as nullable.